generated_codegen.md

```diff
@@ -76,75 +90,107 @@ Uses of `ExecutableWriterHolder` and `ExecutableWriterHolderNoLog` under `src\co
   1231:         ExecutableWriterHolder<DebuggerHeapExecutableMemoryPage> debuggerHeapPageWriterHolder(this, sizeof(DebuggerHeapExecutableMemoryPage));
   1232          DebuggerHeapExecutableMemoryPage *pHeapPageRW = debuggerHeapPageWriterHolder.GetRW();
   1233  #else
-
 ```
 
-`src\coreclr\debug\ee\arm64\walker.cpp`:
+## `src\coreclr\debug\ee\arm64\walker.cpp`:
+
+arm64/macOS: Used for walking and patch skipping - see `controller.cpp` that that is the single caller for this method.
 ```
   178
   179  #if defined(HOST_OSX) && defined(HOST_ARM64)
   180:         ExecutableWriterHolder<UINT_PTR> ripTargetFixupWriterHolder(&m_pSharedPatchBypassBuffer->RipTargetFixup, sizeof(UINT_PTR));
   181          UINT_PTR *pRipTargetFixupRW = ripTargetFixupWriterHolder.GetRW();
   182  #else // HOST_OSX && HOST_ARM64
-
 ```
 
-`src\coreclr\debug\inc\arm64\primitives.h`:
+## `src\coreclr\debug\inc\arm64\primitives.h`:
+
+macOS: Setting of current instruction - `CORDbgSetInstruction()`.
 ```
   152
   153  #if !defined(DBI_COMPILE) && !defined(DACCESS_COMPILE) && defined(HOST_OSX)
   154:     ExecutableWriterHolder<void> instructionWriterHolder((LPVOID)address, sizeof(PRD_TYPE));
   155
   156      TADDR ptraddr = dac_cast<TADDR>(instructionWriterHolder.GetRW());
-
 ```
 
-`src\coreclr\utilcode\loaderheap.cpp`:
+## `src\coreclr\utilcode\loaderheap.cpp`:
+
+DEBUG mechanism to "invalidate" some memory.
 ```
    715              {
    716                  void* pMemRW = pMem;
    717:                 ExecutableWriterHolderNoLog<void> memWriterHolder;
    718                  if (pHeap->IsExecutable())
    719                  {
+```
 
+The loader heap guarantees allocated memory is zero initialized.
+```
    786              {
    787                  void *pResultRW = pResult;
    788:                 ExecutableWriterHolderNoLog<void> resultWriterHolder;
    789                  if (pHeap->IsExecutable())
    790                  {
+```
 
+Set to "invalidate" memory in DEBUG and merge blocks together. This check is only
+for executable memory.
+```
    821                  LoaderHeapFreeBlock *pNextNextBlock = pNextBlock->m_pNext;
    822                  void *pMemRW = pFreeBlock->m_pBlockAddress;
    823:                 ExecutableWriterHolderNoLog<void> memWriterHolder;
    824                  if (pHeap->IsExecutable())
    825                  {
+```
 
+`CommitPages()`
+```
```

Comment:
This generates pages of code with many instances of SubPrecode, FixupPrecode and  CallCountingStub. This block of pages is interleaved with the same sized block of RW pages containing data that the generated stubs use. The m_codePageGenerator is a function that generates the stub code.